### PR TITLE
[WIP] Implement a "scan all" endpoint

### PR DESCRIPTION
**Pull request by caggle**
Using AWS step functions.

### CODE REVIEW

**Comment by caggle** [comment]
I made some pretty good gains in here, but I wouldn't call this an ideal implementation yet. It can be merged without breaking anything though.

In a nutshell, this PR implements:
- A state machine which runs the below tasks sequentially.
image
- All states until the `Waitforresults` state basically kick off a scan as previously implemented.
- `Waitforresults` simply waits for 10 minutes to ensure all the scans run and results are available in the S3 bucket.
- The `Getresults` task returns a pre-signed S3 URL, that contains the results for the host requested.
- The last task `PublishresultURL` sends an email (currently to myself only) containing this link (via SNS). Currently this email looks like the following:
image

TODO items:

- Update documentation to reflect this feature.
- Consider implementing a state to beautify the email message coming from SNS.
- Consider implementing a state to fetch/download Tenable.io scan results in JSON. Currently the downloaded results do **not** include Tenable.io scan results.
- Make the state machine smarter, e.g. instead of running these tasks sequentially, all scans should originate from a port scan as a source of truth. For instance, only run SSH scan if an SSH port is open, only run TLS Observatory scan if a TLS service is offered etc. **This is a longer term implementation.**


@claudijd Feel free to review this as is, if you see any value in doing so.

File: serverless.yml [hunk 1]
```diff
@@ -309,6 +389,7 @@ custom:
     s3BucketName: "vautomator-results"
     vautomatorQ: "vautomator-SQS"
     vautomatorDLQ: "vautomator-DLQ"
+    vautomatorSNS: "Vulnerability Assessment Results"
     ssmKms: beaede06-f6f5-4183-90bf-64a13edcea15
```
Comment (claudijd) [inline serverless.yml]: Is this considered a secret value?

File: serverless.yml [hunk 2]
```diff
@@ -281,6 +353,14 @@ plugins:
...
 resources:
   Resources:
+    MyTopic:
+      Type: AWS::SNS::Topic
+      Properties:
+        DisplayName: ${self:custom.cfg.vautomatorSNS}
+        TopicName: ${self:service}-${self:provider.stage}-vautomator-results-topic
+        Subscription:
+          - Protocol: email
+            Endpoint: culucenk@mozilla.com
```
Comment (claudijd) [inline serverless.yml]: Can we add multiple emails here? Like say you, kang, and myself?

File: serverless.yml (hunk 1 above)
Comment (caggle) [reply serverless.yml]: It's not, this is just the Key ID in the KMS, so SSM uses that Key ID. Afaik there is no way to conceal it in the config, but again, there should not be any need to.

File: serverless.yml (hunk 2 above)
Comment (caggle) [reply serverless.yml]: Based on my research, yes. I will try to deploy with multiple emails.

**Comment by caggle** [comment]
Thanks!

I will merge this now, and will work on the items in `TODO` (and your multiple email endpoints suggestion).

**Pull request merged**